tycho-integration-test/src/four_byte_client.rs

```diff
@@ -0,0 +1,67 @@
+use alloy::transports::http::reqwest;
+use serde::Deserialize;
+use tracing::info;
+
+/// Client for fetching 4byte error signatures.
+///
+/// 4byte.directory is a service that maps 4-byte function signatures to human-readable function
+/// names. See https://www.4byte.directory/ for more.
+///
+/// Note: This client is more reliable for error handling than for fn signature parsing. The latter
+/// contains many spam entries.
+pub struct FourByteClient {
+    url: String,
+    client: reqwest::Client,
+}
+
+#[derive(Debug, Deserialize)]
+struct FourByteResponse {
+    results: Vec<SignatureResult>,
+}
+
+#[derive(Debug, Deserialize)]
+struct SignatureResult {
+    text_signature: String,
+}
+
+impl FourByteClient {
+    pub fn new() -> Self {
+        Self {
+            url: "https://www.4byte.directory/api/v1/signatures/".to_string(),
+            client: reqwest::Client::new(),
+        }
+    }
+
+    pub async fn get_signature(&self, hex_sig: &str) -> Result<String, Box<dyn std::error::Error>> {
+        info!("Fetching signature from 4Byte.directory");
+        let response = self
```

Comment:
uau you don't need any more weird things?? shocked. Why was I getting such weird signatures before with 4bytes? 🤔 was it because it was function signatures and not errors?